encoding/protobuf/error_integration_test.go

```diff
@@ -116,11 +116,13 @@ func TestProtoGrpcServerErrorDetails(t *testing.T) {
 	assert.Equal(t, yarpcerrors.CodeInvalidArgument, st.Code(), "unexpected error code")
 	assert.Equal(t, errorMsg, st.Message(), "unexpected error message")
 	expectedDetails := []interface{}{
-		&types.StringValue{Value: "string value"},
-		&types.Int32Value{Value: 100},
+		&wrappers.StringValue{Value: "string value"},
+		&wrappers.Int32Value{Value: 100},
 	}
 	actualDetails := protobuf.GetErrorDetails(err)
-	assert.Equal(t, expectedDetails, actualDetails, "unexpected error details")
+	if !proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message)) {
+		t.Errorf("unexpected error details")
+	}
```

Comment:
This should be `assert.True(t, proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message))`

Why did you need to change this test?

Reply:
Since we are running in mix mode `gogo <-> google/protobuf` structs(err details in this case) are not equal. This is due to extra added semi-private fields in the structs which are used by the library internally for various usecases.  And hence the `assert.equal` won't work. So using `proto.equal` is more preferable here.